tests/integration/test_remoted/test_agent_communication/test_shared_configuration.py

```diff
@@ -0,0 +1,129 @@
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+def check_push_shared_config(protocol):
+    """Allow to check if the manager sends the shared configuration to agents through remoted
+
+    First, check if the default group configuration file is completely pushed (up message, configuration
+    and close message). Then add the agent to a new group and check if the new configuration is pushed.
+    Also it checks that the same config isn't pushed two times.
+
+    Args:
+        protocol (str): It can be UDP or TCP.
+
+    Raises:
+        TimeoutError: If agent does not receive the manager ACK message in the expected time.
+    """
+
+    # Create agent and sender object with default parameters
+    agent = ag.Agent(**agent_info)
+
+    # Sleep to avoid ConnectionRefusedError
+    sleep(1)
+
+    sender = ag.Sender(agent_info['manager_address'], protocol=protocol)
+
+    # Activate receives_messages modules in simulated agent.
+    agent.set_module_status('receive_messages', 'enabled')
+
+    # Run injector with only receive messages module enabled
+    injector = ag.Injector(sender, agent)
+    try:
+        injector.run()
+
+        # Wait until remoted has loaded the new agent key
+        rd.wait_to_remoted_key_update(wazuh_log_monitor)
+
+        # Send the start-up message
+        sender.send_event(agent.startup_msg)
+        sender.send_event(agent.keep_alive_event)
+
+        # Check up file (push start) message
+        rd.check_agent_received_message(agent.rcv_msg_queue, r'#!-up file \w+ merged.mg', timeout=10,
+                                        error_message="initial up file message not received")
+
+        # Check agent.conf message
+        rd.check_agent_received_message(agent.rcv_msg_queue, '#default', timeout=10,
+                                        error_message="agent.conf message not received")
+        # Check close file (push end) message
+        rd.check_agent_received_message(agent.rcv_msg_queue, 'close', timeout=10,
+                                        error_message="initial close message not received")
+        sender.send_event(agent.keep_alive_event)
+        # Check that push message doesn't appear again
+        with pytest.raises(TimeoutError):
+            rd.check_agent_received_message(agent.rcv_msg_queue, r'#!-up file \w+ merged.mg', timeout=5)
+            raise AssertionError("Same shared configuration pushed twice!")
+
+        # Add agent to group and check if the configuration is pushed.
+        subprocess.run(["/var/ossec/bin/agent_groups", "-q", "-a", "-i", agent.id, "-g", "testing_group"])
+        sender.send_event(agent.keep_alive_event)
+        rd.check_agent_received_message(agent.rcv_msg_queue, '#!-up file .* merged.mg', timeout=10,
+                                        error_message="New group shared config not received")
+
+    finally:
+        injector.stop_receive()
```

Comment:
I prefer to move these kinds of functions to the module to reduce the logic implemented in the tests. Also, keep in mind you need to use the variable `WAZUH_PATH`.

Reply:
https://github.com/wazuh/wazuh-qa/pull/1115/commits/750ca4b0fa26133d94595077dcb1039b89a315ce